classes/fields/avatar.php

```diff
@@ -88,30 +88,151 @@ public function get_avatar( $avatar, $id_or_email, $size, $default = '', $alt =
 			}
 		}
 
-		$user_id = 0;
+		$user_id = $this->get_avatar_user_id( $id_or_email );
 
-		if ( is_numeric( $id_or_email ) && 0 < $id_or_email ) {
-			$user_id = (int) $id_or_email;
-		} elseif ( is_object( $id_or_email ) && isset( $id_or_email->user_id ) && 0 < $id_or_email->user_id ) {
-			$user_id = (int) $id_or_email->user_id;
-		} elseif ( is_object( $id_or_email ) && isset( $id_or_email->ID ) && isset( $id_or_email->user_login ) && 0 < $id_or_email->ID ) {
-			$user_id = (int) $id_or_email->ID;
-		} elseif ( ! is_object( $id_or_email ) && false !== strpos( $id_or_email, '@' ) ) {
-			$_user = get_user_by( 'email', $id_or_email );
+		if ( 0 < $user_id && ! empty( PodsMeta::$user ) ) {
+			$avatar_cached = pods_cache_get( $user_id . '-' . $size, 'pods_avatars' );
 
-			if ( ! empty( $_user ) ) {
-				$user_id = (int) $_user->ID;
+			if ( ! empty( $avatar_cached ) ) {
+				$avatar = $avatar_cached;
+			} else {
+
+				$user_avatar = $this->get_avatar_id( $user_id );
+
+				if ( ! empty( $user_avatar ) ) {
+					$attributes = array(
+						'alt'   => '',
+						'class' => 'avatar avatar-' . $size . ' photo',
+					);
+
+					if ( ! empty( $alt ) ) {
+						$attributes['alt'] = $alt;
+					}
+
+					$user_avatar = pods_image( $user_avatar, array( $size, $size ), 0, $attributes );
+
+					if ( ! empty( $user_avatar ) ) {
+						$avatar = $user_avatar;
+
+						pods_cache_set( $user_id . '-' . $size, $avatar, 'pods_avatars' );
+					}
+				}
 			}
 		}
 
+		return $avatar;
+	}
+
+	/**
+	 * Take over the avatar data served from WordPress
+	 *
+	 * @since 2.7.22
+	 *
+	 * @param mixed $id_or_email The Gravatar to retrieve. Accepts a user ID, Gravatar MD5 hash,
+	 *                           user email, WP_User object, WP_Post object, or WP_Comment object.
+	 * @param array $args {
```

Comment:
:no_entry_sign: Doc comment for parameter $args does not match actual variable name $id_or_email

<sub>hash: 2966668ad1b8a89012bf5779b4d00257</sub>